### PR TITLE
Terraform Upgrade in Automate HA

### PR DESCRIPTION
**Pull request by schaudha**
### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-7297

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-backend-deployment/habitat/bin/prepare-a2-deployment-workspace [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash -xe
 
 echo "A2 deployment workspace will be created in ${PWD}/a2_deploy_workspace.  Please enter your sudo password if prompted"
-sudo rsync -avz  $(hab pkg path chef/automate-ha-deployment)/workspace/ a2_deploy_workspace
+sudo rsync -avz  $(hab pkg path schaudha/automate-ha-deployment)/workspace/ a2_deploy_workspace
```
Comment (Dmaddu) [inline components/automate-backend-deployment/habitat/bin/prepare-a2-deployment-workspace:4]: we should not change the origin from `chef` to your origin

File: components/automate-backend-deployment/habitat/hooks/install [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #!{{pkgPathFor "core/bash"}}/bin/bash
 
 hab pkg binlink chef/inspec -f
-hab pkg binlink core/terraform -f
+hab pkg binlink schaudha/terraform/1.5.7/20230921122623 -f
 hab pkg binlink core/jq-static -f
-hab pkg binlink chef/automate-ha-cluster-ctl -f
+hab pkg binlink jayvikramsharma/automate-ha-cluster-ctl/0.1.0/20230927115358 -f
```
Comment (Dmaddu) [inline components/automate-backend-deployment/habitat/hooks/install:6]: version pin is not recommended. Please remove the version and change the origin

File: components/automate-backend-deployment/habitat/plan.sh [hunk 3]
```diff
@@ -16,14 +16,14 @@ pkg_deps=(
   core/gawk
   core/gzip
   core/jq-static
-  core/openssh
-  core/openssl
+  core/openssh/7.9p1/20220311132702
+  core/openssl/1.0.2zb/20220311111046
```
Comment (Dmaddu) [inline components/automate-backend-deployment/habitat/plan.sh:20]: in the latest dev, we have upgraded to openssl 1.0.2zf version, please test with that version. 
Never pin a version for packages

**Review by Dmaddu — CHANGES_REQUESTED** [review]

**Review by punitmundra — COMMENTED** [review]
any reason these certificate got regenerated ?

File: components/automate-backend-deployment/habitat/bin/prepare-a2-deployment-workspace (hunk 1 above)
Comment (schaudha) [reply components/automate-backend-deployment/habitat/bin/prepare-a2-deployment-workspace:4]: Hi @Dmaddu, the changes were made to test the terraform version after making changes in the code. I will revert it back to chef.

File: components/automate-backend-deployment/habitat/plan.sh (hunk 3 above)
Comment (schaudha) [reply components/automate-backend-deployment/habitat/plan.sh:20]: Hi @Dmaddu, the changes were made to test the terraform version after making changes in the code. I will revert it back to chef.

File: components/automate-cli/cmd/chef-automate/version.go [hunk 4]
```diff
@@ -165,27 +166,47 @@ func printUpgradeStatus() error {
 }
 
 func init() {
-	versionCmd.Flags().BoolVarP(&VersionCommandFlags.verbose, "verbose", "v", false, "Show additional version information")
```
Comment (punitmundra) [inline components/automate-cli/cmd/chef-automate/version.go:168]: is this change is the part of some other PR ?

Comment (schaudha) [reply components/automate-cli/cmd/chef-automate/version.go:168]: Looks like it. This is not from this PR.

**Review by punitmundra — CHANGES_REQUESTED** [review]
@schaudha : can you remove the other change which are not the part of terrform upgrade

**Review by ArvinthC3000 — COMMENTED** [review]
Rebase the branch with main

File: components/automate-cluster-ctl/habitat/plan.sh [hunk 5]
```diff
@@ -18,11 +18,11 @@ pkg_deps=(
...
   core/jq-static
   core/openssh
-  core/openssl
+  core/openssl/1.0.2zb/20220311111046
   core/make
   core/curl
-  core/rsync
-  core/terraform/0.14.8/20210826165930
+  core/rsync/3.2.3/20220817235706
+  schaudha/terraform/1.5.7/20230921122623
```
Comment (punitmundra) [inline components/automate-cluster-ctl/habitat/plan.sh:25]: can we make the final changes for the terrafrom package ?

File: components/automate-cluster-ctl/habitat/plan.sh [hunk 6]
```diff
@@ -18,11 +18,11 @@ pkg_deps=(
   core/gzip
   core/jq-static
   core/openssh
-  core/openssl
+  core/openssl/1.0.2zb/20220311111046
   core/make
   core/curl
-  core/rsync
-  core/terraform/0.14.8/20210826165930
+  core/rsync/3.2.3/20220817235706
```
Comment (punitmundra) [inline components/automate-cluster-ctl/habitat/plan.sh:24]: is  the rsync pkg got updated in builder and with the new pkg with build is failing then pining the pkg is fine other wise we need to unpin the pkg and re-test the senariao

Comment (schaudha) [reply components/automate-cluster-ctl/habitat/plan.sh]: The pkg was used for testing purposes. I have tested the same with the default core/rsync package.

File: components/automate-cluster-ctl/habitat/plan.sh (hunk 5 above)
Comment (schaudha) [reply components/automate-cluster-ctl/habitat/plan.sh]: I have pushed the changes which have the new terraform core/terraform1 package.

File: components/automate-cluster-ctl/habitat/plan.sh [hunk 7]
```diff
@@ -22,8 +22,8 @@ pkg_deps=(
   core/make
   core/curl
   core/rsync
-  core/terraform/0.14.8/20210826165930
-  core/hab/1.6.521/20220603154827
+  core/terraform1
+  core/hab
```
Comment (punitmundra) [inline components/automate-cluster-ctl/habitat/plan.sh:26]: @schaudha : why you removed the hab pin version ?

Comment (schaudha) [reply components/automate-cluster-ctl/habitat/plan.sh]: @punitmundra It was unknowingly. I just reverted to the pinned version.

File: components/automate-backend-deployment/habitat/hooks/install (hunk 2 above)
Comment (schaudha) [reply components/automate-backend-deployment/habitat/hooks/install]: Removed the pinned version and reverted the origin.

File: components/automate-backend-deployment/habitat/plan.sh (hunk 3 above)
Comment (schaudha) [reply components/automate-backend-deployment/habitat/plan.sh]: Removed the pinned version and reverted the origin of the package to its original.

**Review by punitmundra — APPROVED** [review]

**Review by jayvikramsharma1 — APPROVED** [review]

**Pull request merged**